app/controllers/activity_sessions_controller.rb

```diff
@@ -8,14 +8,23 @@ class ActivitySessionsController < ApplicationController
 
   def play
     if @activity_session.state == "finished"
-      newSession = ActivitySession.create(
-       state: "unstarted",
-       user_id: @activity_session.user_id,
-       activity_id: @activity_session.activity_id,
-       classroom_activity_id: @activity_session.classroom_activity_id,
-       is_retry: true
-      )
-      redirect_to play_activity_session_path(newSession)
+      started_session = ActivitySession.where(
+        user_id: @activity_session.user_id,
+        activity_id: @activity_session.activity_id
+        ).last
+      if started_session && started_session.started_at > @activity_session.completed_at
+        binding.pry
+        redirect_to play_activity_session_path(started_session)
+      else
+        newSession = ActivitySession.create(
```

Comment:
I know this is kind of pedantic and you didn't write it, but can you change this to new_session when you remove the binding.